pool/web/handler/admin_experiments.ml

```diff
@@ -100,33 +100,46 @@ let experiment_message_templates database_label experiment_id =
 let index req =
   let open Utils.Lwt_result.Infix in
   let error_path = "/admin/dashboard" in
-  let result ({ Pool_context.database_label; user; _ } as context) =
-    let find_actor =
-      Pool_context.Utils.find_authorizable ~admin_only:true database_label user
-    in
-    let find_experiments actor =
-      let open Experiment in
-      let query =
-        Query.from_request
-          ~searchable_by
-          ~sortable_by
-          ~default:Experiment.default_query
-          req
-      in
-      find_all
-        ~query
-        ~actor
-        ~permission:Guard.Access.index_permission
-        database_label
+  let active_navigation = "/admin/experiments" in
+  HttpUtils.extract_happy_path ~src req
+  @@ fun ({ Pool_context.database_label; user; _ } as context) ->
+  let find_actor =
+    Pool_context.Utils.find_authorizable ~admin_only:true database_label user
+  in
+  let find_experiments actor =
+    let open Experiment in
+    let query =
+      Query.from_request
+        ~searchable_by
+        ~sortable_by
+        ~default:Experiment.default_query
+        req
     in
-    find_actor
-    |>> find_experiments
-    >|+ Page.Admin.Experiments.index context
-    >>= create_layout ~active_navigation:"/admin/experiments" req context
-    >|+ Sihl.Web.Response.of_html
-    >|- fun err -> err, error_path
+    find_all
+      ~query
+      ~actor
+      ~permission:Guard.Access.index_permission
+      database_label
   in
-  result |> HttpUtils.extract_happy_path ~src req
+  let* actor = find_actor >|- fun err -> err, error_path in
+  let%lwt experiments, query = find_experiments actor in
+  let page =
+    Page.Admin.Experiments.Index_view.make
+      ~with_search:(not (HttpUtils.is_hx_request req))
+      context
+      experiments
+      query
+  in
+  if HttpUtils.is_hx_request req
```

Comment:
**note**. I know this pattern will reocurr in other tables, but I'll wait until we have that code done to see how if a nice abstraction emerges. I'd advice against pushing this _now_ into `HttpUtils.extract_happy_path` since that is already rather hard to read.